From 427962a4c23bd7f8426646f246dc579a222da3b6 Mon Sep 17 00:00:00 2001
From: Wang Zaikuo <zaikuo.wang@intel.com>
Date: Wed, 9 May 2018 10:28:22 +0800
Subject: [PATCH 1902/2345] media: intel-ipu4: psys: correct resource
 definition

Message for Open Source:
The resource definition array: ipu_fw_psys_cell_mem[][] is not align to
the hardware spec.

This patch fix above issue and align the ipu_fw_psys_cell_mem
array element

Message for Internal:
[Issue/Feature]
For FW ABI flow, the ipu_fw_psys_cell_mem[][] would be
used and should be align to FW api definition:
vied_nci_cell_mem; But currently the second dimension
size is not same to FW definition.

This issue only impact ipu4p because FW ABI flow is enabled
and used on ipu4p
FW API flow uses resource definitions provided by FW directly.

[Root Cause/Changes]
ipu4/ipu-platform-resources.h
ipu4/ipu4-fw-resources.c
ipu4/ipu4-resources.c

Change-Id: I8095aa27a6b4bae1a9086c36a187db6002940781
Signed-off-by: Wang Zaikuo <zaikuo.wang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 .../media/pci/intel/ipu4/ipu-platform-resources.h  |  2 +-
 drivers/media/pci/intel/ipu4/ipu4-fw-resources.c   |  2 +-
 drivers/media/pci/intel/ipu4/ipu4-resources.c      | 27 ++++++++++++++++++----
 3 files changed, 24 insertions(+), 7 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu-platform-resources.h b/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
index 784a030..68e26a4 100644
--- a/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
+++ b/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
@@ -224,7 +224,7 @@ extern const u32 ipu_fw_psys_cell_types[];
 extern const u16 ipu_fw_num_dev_channels[];
 extern const u16 ipu_fw_psys_mem_size[];
 extern const enum ipu_mem_id ipu_fw_psys_cell_mem
-	[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_DATA_MEM_TYPE_ID];
+	[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_MEM_TYPE_ID];
 extern const struct ipu_fw_resource_definitions *res_defs;
 
 #endif /* IPU_PLATFORM_RESOURCES_H */
diff --git a/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c b/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
index 2229f03..9177c3e 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
@@ -17,7 +17,7 @@ static const struct ipu_fw_resource_definitions default_defs = {
 
 	.num_dfm_ids = IPU_FW_PSYS_N_DEV_DFM_ID,
 
-	.cell_mem_row = IPU_FW_PSYS_N_DATA_MEM_TYPE_ID,
+	.cell_mem_row = IPU_FW_PSYS_N_MEM_TYPE_ID,
 	.cell_mem = &ipu_fw_psys_cell_mem[0][0],
 
 	.process.ext_mem_id = offsetof(struct ipu_fw_psys_process,
diff --git a/drivers/media/pci/intel/ipu4/ipu4-resources.c b/drivers/media/pci/intel/ipu4/ipu4-resources.c
index 0bed8da..fba8c65 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-resources.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-resources.c
@@ -75,107 +75,124 @@ const u16 ipu_fw_psys_mem_size[IPU_FW_PSYS_N_MEM_ID] = {
 };
 
 const enum ipu_mem_id
-ipu_fw_psys_cell_mem[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_DATA_MEM_TYPE_ID] = {
+ipu_fw_psys_cell_mem[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_MEM_TYPE_ID] = {
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_DMEM0_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_DMEM1_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_DMEM2_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_VMEM4_ID,
 	 IPU_FW_PSYS_DMEM4_ID,
 	 IPU_FW_PSYS_VMEM0_ID,
-	 IPU_FW_PSYS_BAMEM0_ID
+	 IPU_FW_PSYS_BAMEM0_ID,
+	 IPU_FW_PSYS_PMEM0_ID
 	},
 	{
 	 IPU_FW_PSYS_VMEM4_ID,
 	 IPU_FW_PSYS_DMEM5_ID,
 	 IPU_FW_PSYS_VMEM1_ID,
-	 IPU_FW_PSYS_BAMEM1_ID
+	 IPU_FW_PSYS_BAMEM1_ID,
+	 IPU_FW_PSYS_PMEM1_ID
 	},
 	{
 	 IPU_FW_PSYS_VMEM4_ID,
 	 IPU_FW_PSYS_DMEM6_ID,
 	 IPU_FW_PSYS_VMEM2_ID,
-	 IPU_FW_PSYS_BAMEM2_ID
+	 IPU_FW_PSYS_BAMEM2_ID,
+	 IPU_FW_PSYS_PMEM2_ID,
 	},
 	{
 	 IPU_FW_PSYS_VMEM4_ID,
 	 IPU_FW_PSYS_DMEM7_ID,
 	 IPU_FW_PSYS_VMEM3_ID,
-	 IPU_FW_PSYS_BAMEM3_ID
+	 IPU_FW_PSYS_BAMEM3_ID,
+	 IPU_FW_PSYS_PMEM3_ID,
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	},
 	{
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
 	 IPU_FW_PSYS_N_MEM_ID
 	}
 };
-- 
2.7.4

